actors/builtin/miner/miner_state.go

```diff
@@ -320,6 +323,20 @@ func (st *State) DeleteSectors(store adt.Store, sectorNos *abi.BitField) error {
 	return err
 }
 
+func (st *State) DeductPledges(store adt.Store, sectorNos *abi.BitField) error {
+	return sectorNos.ForEach(func(sectorNo uint64) error {
+		sector, found, err := st.GetSector(store, abi.SectorNumber(sectorNo))
```

Comment:
Blocking: Unfortunately this is hiding a potentially huge performance issue inside the state class. Loading an arbitrary number of sector infos is unreasonable. This will need to be replaced as part of #411, using a similar technique, and the logic will need to move. In order to make that easier later, please let's not put it inside the state class. cc @Stebalien 

- Remove this method, do the work in the actor code and then use `AddInitialPledge(pledgeToRemove.Neg())`.
- Remove the call site from `removeTerminatedSectors` and pull up to `terminateSectors`. That method _already_ loads all these infos, so accumulate the pledge delta there and then call the cheap state mutation